pypeit/core/extract.py

```diff
@@ -1842,6 +1875,23 @@ def ech_objfind(image, ivar, slitmask, slit_left, slit_righ, order_vec, maskslit
     for iord in range(norders):
         slit_spat_pos[iord, :] = (np.interp(slit_spec_pos, spec_vec, slit_left[:,iord]), np.interp(slit_spec_pos, spec_vec, slit_righ[:,iord]))
 
+
+    # Hand prep
+    if hand_extract_dict is not None:
+        f_spats = []
+        for ss, spat, spec in zip(range(len(hand_extract_dict['hand_extract_spec'])),
+                              hand_extract_dict['hand_extract_spat'],
+                              hand_extract_dict['hand_extract_spec']):
+            # Find the input slit
+            ispec = int(spec)
```

Comment:
This should be np.clip(np.round(spec),0,nspec-1) or something like that. Always round instead of int, which is equivalent to a floor.

Reply:
Fixed by me.